tree/dataframe/test/dataframe_nodes.cxx

```diff
@@ -86,3 +87,24 @@ TEST(RDataFrameNodes, DoubleEvtLoop)
    for (auto &f : files)
       gSystem->Unlink(f.c_str());
 }
+
+// ROOT-9736
+TEST(RDataFrameNodes, InheritanceOfCustomColumns)
```

Comment:
Can you make it clearer where the feature is used (cursory look I see TH1F vs TStatistic which don't seem to be linked)? (i.e. what am I missing?)

Reply:
I don't understand the comment.
I generate a column b which is of type TH1F and read it as a column of TH1.
I generate a column x of TStatistic, dump on disk the obtained dataset, create a RDF on top of it, read the column x as TObject.

Reply:
> a cursory read would lead to the 'but-of-course-you-can-call a function taking a TH1* with a TH1F* star' 

Exactly! users expect it to be possible, but it wasn't until now